src/main/java/net/snowflake/client/jdbc/cloud/storage/SnowflakeGCSClient.java

```diff
@@ -557,6 +557,8 @@ public void uploadWithPresignedUrlWithoutConnection(
     logger.debug("Starting upload");
     uploadWithPresignedUrl(
         networkTimeoutInMilli,
+        0, // auth timeout
```

Comment:
NIT: Introduce a static int for default auth timeout like `DEFAULT_HTTP_CLIENT_SOCKET_TIMEOUT` instead of using `0`?